apps/ejabberd/src/ejabberd_admin.erl

```diff
@@ -155,6 +166,59 @@ commands() ->
 %%% Server management
 %%%
 
+-spec join_cluster(string()) -> {ok, string()} | {pang, string()} | {mnesia_error, string()} | {error, string()}.
+join_cluster(NodeString) ->
+    NodeAtom = list_to_atom(NodeString),
+    NodeList = mnesia:system_info(running_db_nodes),
+    case lists:member(NodeAtom, NodeList) of
+        true ->
+            String = io_lib:format("The node ~s has already joined the cluster~n", [NodeString]),
+            {ok, String};
+        _ ->
+            do_join_cluster(NodeAtom)
+    end.
+
+do_join_cluster(Node) ->
+    try mongoose_cluster:join(Node) of
+        ok ->
+            String = io_lib:format("You have successfully joined the node ~p to
+      the cluster with node member ~p~n", [node(), Node]),
+            {ok, String}
+    catch
+        error:pang ->
+            String = io_lib:format("Timeout while attempting to connect to node ~s~n", [Node]),
+            {pang, String};
+        error:{cant_get_storage_type, {T, E, R}} ->
+            String = io_lib:format("Cannot get storage type for table ~p~n. Reason: ~p:~p", [T, E, R]),
+            {mnesia_error, String};
+        E:R ->
+            {error, {E, R}}
+    end.
+
+-spec leave_cluster() -> {ok, string()} | {error, term()} | {not_in_cluster, string()}.
+leave_cluster() ->
+    NodeList = mnesia:system_info(running_db_nodes),
+    ThisNode = node(),
+    case NodeList of
+        [ThisNode] ->
```

Comment:
If there is more than one node the case would match here [1] and it would mean that node is in any cluster and can leave it. Otherwise [2] there is no cluster to leave because `mnesia:system_info(running_db_nodes)` returned just the local node
**However you're right about the lacking tests!**

[1]https://github.com/esl/MongooseIM/blob/rel-1.7-mongoose-cluster/apps/ejabberd/src/ejabberd_admin.erl#L206
[2]https://github.com/esl/MongooseIM/blob/rel-1.7-mongoose-cluster/apps/ejabberd/src/ejabberd_admin.erl#L203